lib/active_storage_validations/dimension_validator.rb

```diff
@@ -51,64 +57,65 @@ def validate_each(record, attribute, _value)
       def validate_each(record, attribute, _value)
         return true unless record.send(attribute).attached?
 
+        flat_options = process_options(record)
         files = Array.wrap(record.send(attribute))
         files.each do |file|
           # Analyze file first if not analyzed to get all required metadata.
           file.analyze; file.reload unless file.analyzed?
           metadata = file.metadata rescue {}
-          next if is_valid?(record, attribute, metadata)
+          next if is_valid?(record, attribute, metadata, flat_options)
           break
         end
       end
     end
 
 
-    def is_valid?(record, attribute, file_metadata)
+    def is_valid?(record, attribute, file_metadata, flat_options)
```

Comment:
Same here, you already have the record. Less arguments = better :)

Reply:
Agreed and changed :)